### PR TITLE
Fix errors with `detektGenerateConfig`

### LINKED ISSUE #4843: `detektGenerateConfig` fails
## Expected Behavior
If I run `./gradlew detektGenerateConfig` the configuration file should be created and gradle should not fail.

## Observed Behavior
The file is created, but gradle fails: https://ge.detekt.dev/s/lpgeopyjrd46k/failure?expanded-stacktrace=WyIwIiwiMC03LTktMTEiXQ&focused-stack-frame=0-7-9-11-0

## Steps to Reproduce
In this project:
```
rm config/detekt/detekt.yml
./gradlew detektGenerateConfig
```

## Context
This is a long standing bug. I thought it was reported but I can't find it.

## Your Environment
* Version of detekt used: 1.20.0 / main
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task): https://ge.detekt.dev/s/lpgeopyjrd46k
* Link to your project (if it's a public repository): https://github.com/detekt/detekt

### PR DESCRIPTION
**Pull request by BraisGabin**
I recomment to read #4843 to understand all the context for this solution.

TL;DR `detektGenerateConfig` runs in a race condition and tries to create the same file multiple times and crashes. With this solution we ansure that this task is always run in serial to avoid it.

This PR also have 2 commits with little refactors over `DetektGenerateConfigTask`. I didn't want to fall into the rabbit hole so I didn't refactor anymore. But if someone wants: It seems that we are doing more work than needed here. The providers and properties are executed multiple times without, I think, a good reason.

Fixes #4843

### CODE REVIEW

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt [hunk 1]
```diff
@@ -45,34 +44,37 @@ abstract class DetektGenerateConfigTask @Inject constructor(
...
 
-    private val configurationToUse: ConfigurableFileCollection
+    private val configurationToUse: File
         get() = if (config.isEmpty) {
-            objects.fileCollection().from(defaultConfigPath)
+            defaultConfigPath.toFile()
         } else {
-            config
+            config.last()
         }
```
Comment (chao2zhang) [inline detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt:53]: Just to confirm that these changes are mainly for refactoring.

**Review by chao2zhang — APPROVED** [review]

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt (hunk 1 above)
Comment (BraisGabin) [reply detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt:53]: Yes

**Review by schalkms — APPROVED** [review]
